### PR TITLE
Add support for `smath::log()`, `smath::logX()`

### PR DESCRIPTION
**Pull request by Nava2**
* Add support for `smath::log()`, matches `std::log()`
* Add support for `smath::log10()`, matches `std::log10()`
* Add support for `smath::log2()`, matches `std::log2()`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 `static_math` is a C++ library dedicated to compile time mathematics.
 
+[![Build Status](https://travis-ci.org/Morwenn/static_math.svg?branch=master)](https://travis-ci.org/Morwenn/static_math)
```
Comment (Morwenn) [inline README.md]: Could you put it right on the left of the license badge instead, please? :)

Comment (Nava2) [inline README.md]: Done

**Comment by Nava2** [comment]
I forgot to mention this, but the implementation is not perfect. 

Due to the fact that templates can only recurse so low, the Taylor series expansion can only be run so deep. Thus, I had to use an approximation method where-in you use `x = A * 10^n` and `ln(x) = ln(A) + n * ln10`. 

The numbers are close, but not perfect (they're not within a single epsilon :-1:).

**Comment by Nava2** [comment]
The problem is that for big numbers, it starts going too deep into the `step count` of the `constexpr` for the compilers.

**Comment by Morwenn** [comment]
Hum, ok. I will merge it as is for now and you can always come back later with improvements if you feel like it anyway :)

**Comment by Nava2** [comment]
Yeah, it became more of an academic thing. I can't use the library anyway, thanks to MSVC.

**Comment by Morwenn** [comment]
Maybe someday in the future :p

Just to be curious: what did you want to use it for?

**Comment by Nava2** [comment]
Just better math in my thesis work. I have to do a bunch of arbitrary low-level math functions that I know are simple enough for static compile-time math.

**Comment by Morwenn** [comment]
Oh, ok. It's not the first time someone contributes to the library, but except for the most basic integer functions, I never needed to do math at compile time either, so I've always wondered why people found the library either. I guess that it might be for same reason too.

**Pull request merged**